circle.yml

```diff
@@ -5,8 +5,8 @@ machine:
     version:
       oraclejdk8
   environment:
-    _JAVA_OPTIONS: "-Xmx512m"
-    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx512m -XX:+HeapDumpOnOutOfMemoryError"'
+    _JAVA_OPTIONS: "-Xmx384M"
+    GRADLE_OPTS: '-Xss1024K -XX:+CMSClassUnloadingEnabled -XX:InitialCodeCacheSize=32M -XX:CodeCacheExpansionSize=1M -XX:CodeCacheMinimumFreeSpace=1M -XX:ReservedCodeCacheSize=150M -XX:MinMetaspaceExpansion=1M -XX:MaxMetaspaceExpansion=8M -XX:MaxMetaspaceSize=128M -XX:MaxDirectMemorySize=96M -XX:CompressedClassSpaceSize=128M'
```

Comment:
Would definitely like to get some details on how much heap, meta, and code cache we're actually using for builds. Should probably just enable GC logs for gradle and he tests and capture those in build output.

For circle testing purposes, you may want to adjust the cores and memory used by C* similar to https://github.com/openzipkin/zipkin/pull/1325